source/common/quic/envoy_quic_server_stream.cc

```diff
@@ -53,12 +54,18 @@ void EnvoyQuicServerStream::encodeHeaders(const Http::ResponseHeaderMap& headers
     IS_ENVOY_BUG("encodeHeaders is called on write-closed stream.");
     return;
   }
+  auto modified_headers = Http::createHeaderMap<Http::ResponseHeaderMapImpl>(headers);
+  // Transform Response from H1 to H3
```

Comment:
Please update this comment and the one on line 214 as per the changes in the client_stream